core/src/entrypoint/entity.rs

```diff
@@ -342,150 +275,225 @@ impl EntityBuilder {
 mod tests {
     use super::*;
 
-    #[test]
-    fn aristotle() {
+    #[macro_export]
+    macro_rules! expect {
+        ($($t:tt)*) => {
+            |res| ::insta::assert_display_snapshot!(res, $($t)*)
+        };
+    }
+
+    fn fmt_span(s: &Span) -> String {
+        s.links.iter().rfold(s.text.clone(), |mut acc, link| {
+            if acc.len() <= link.start || acc.len() <= link.end {
+                return acc;
+            }
+            let start = link.start;
+            let end = link.end;
+            if link.target == acc[start..end] {
+                acc.insert_str(end, "]]");
+                acc.insert_str(start, "[[");
+            } else {
+                acc.insert_str(end, &format!("]({})", link.target));
+                acc.insert(start, '[');
+            }
+            acc
+        })
+    }
+
+    fn ellipsis(s: &str) -> String {
+        if s.len() > 128 {
+            format!("{}…", &s[0..128])
+        } else {
+            s.to_string()
+        }
+    }
+
+    fn check_abstract(title: &str, src: &str, f: impl FnOnce(String)) {
```

Comment:
Can we have a more descriptive name than `f` here? It's not obvious without reading about insta what the closure does

Reply:
Renamed to `expect` (the name used in rust-analyzer 😉) and added a short doc string.